### ISSUE TITLE
Cannot instantiate functools.partial without errors 

### ISSUE THREAD

**Issue by NeilGirdhar** [python/mypy#9124]
When I run MyPy 0.782 (`mypy --disallow-untyped-calls --disallow-untyped-defs --disallow-incomplete-defs`) on this file:
```
from functools import partial


class Partial(partial):
    pass
```
it reports:
```
a.py:4: error: Missing type parameters for generic type "partial"
```
But, if I try to provide the generic parameter shown in the typeshed:
```
class Partial(partial[int]):
    pass
```
Then running the code on Python 3.8.2 says
```
TypeError: 'type' object is not subscriptable
```
How am I supposed to instantiate `partial` with [its generic parameter][1]?  

  [1]: https://github.com/python/typeshed/blob/master/stdlib/3/functools.pyi#L62

**Comment by hauntsaninja** [comment]
As far as I know, there isn't a clean way. The not-so-clean way I know of is to conditionally use a metaclass. This post goes through the details: https://stackoverflow.com/a/48554601 

And as always, type ignore or `--allow-any-generics` or whatever is the get-out-of-my-way solution.

**Comment by NeilGirdhar** [comment]
@hauntsaninja Thanks!   Is this going to be fixed eventually?

**Comment by hauntsaninja** [comment]
I don't think CPython plans on making `functools.partial` inherit from `Generic`. If you don't want the specificity that treating `functools.partial` as generic gets you during type checking, you can use the mypy option `--allow-any-generics` and then mypy won't complain about your first example.

The more common way one encounters this problem is through annotations (as opposed to inheritance), and that can be made to go away by using `from __future__ import annotations`. But specifying base classes is something that necessarily affects both runtime and type checking (and in practice this is only an issue for a handful of classes), so I don't see what more of a fix could look like.

**Comment by NeilGirdhar** [comment]
Thanks, yeah, the "fix" is for the standard library to absorb the annotations from the typeshed.  Hopefully, this eventually happens because switching the base class out is ugly.

**Comment by hauntsaninja** [comment]
As it happens, this is fixed in Python 3.9, as part of PEP 585

**Comment by NeilGirdhar** [comment]
@hauntsaninja awesome, thanks for letting me know.
